### PR TITLE
feat: accept list of ints in gpu_ids param

### LINKED ISSUE #518: GPU not found error. pynvml.nvml.NVMLError_NotSupported: Not Supported
* CodeCarbon version: 2.3.4
* Python version: 3.10.13
* Operating System: Linux Ubuntu 22.04

I want to measure the cpu energy consumption of my python code using code carbon. My gpu version is Nvidia GeForce MX250 which does not allow energy monitoring. When I run the code I get this error because the codecarbon code is trying to find the gpu.
Screenshot from 2024-03-21 02-16-31

image

But when I run the same code in kaggle notebook with the same codecarbon version and python version, it is able to monitor only the cpu energy consumption when is disabled. Why is this the case?

This is the output from kaggle.
Screenshot from 2024-03-21 02-20-47

This is the code that I am running.

import tensorflow as tf

from codecarbon import EmissionsTracker

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential(
    [
        tf.keras.layers.Flatten(input_shape=(28, 28)),
        tf.keras.layers.Dense(128, activation="relu"),
        tf.keras.layers.Dropout(0.2),
        tf.keras.layers.Dense(10),
    ]
)

loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

model.compile(optimizer="adam", loss=loss_fn, metrics=["accuracy"])

tracker = EmissionsTracker(gpu_ids=[])
tracker.start()
model.fit(x_train, y_train, epochs=10)
emissions: float = tracker.stop()
print(emissions)

Please help me identify what the issue is?

```

### PR DESCRIPTION
**Pull request by inimaz**
Allow users to specify a list of ints for the gpu_ids parameter
Closes #518 

# Before this PR
```
tracker = EmissionsTracker(gpu_ids="1,3,5")
```
# After this PR
```
tracker = EmissionsTracker(gpu_ids="1,3,5")
```
or
```
tracker = EmissionsTracker(gpu_ids=[1,3,5])
```

### CODE REVIEW

**Review by benoit-cty — APPROVED** [review]
I think the original format is from [CUDA_VISIBLE_DEVICES](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars) so it is nice to keep it.
